### PR TITLE
Map armor trims and related things

### PR DESCRIPTION
**Pull request by MattiDragon**
Open to suggestions for more related things to map

### CODE REVIEW

**Comment by MattiDragon** [comment]
Made this a draft as screen stuff is still WIP

**Comment by MattiDragon** [comment]
I think I'll stop adding things now. Ready for review

File: mappings/net/minecraft/LegacySmithingScreen.mapping [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CLASS net/minecraft/class_8065 net/minecraft/LegacySmithingScreen
```
Comment (haykam821) [inline mappings/net/minecraft/LegacySmithingScreen.mapping]: This class is not in the correct package.

File: mappings/net/minecraft/LegacySmithingScreenHandler.mapping [hunk 2]
```diff
@@ -0,0 +1,24 @@
+CLASS net/minecraft/class_8050 net/minecraft/LegacySmithingScreenHandler
```
Comment (haykam821) [inline mappings/net/minecraft/LegacySmithingScreenHandler.mapping]: This class is also not in the correct package.

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/item/trim/ArmorTrimPattern.mapping [hunk 3]
```diff
@@ -0,0 +1,5 @@
+CLASS net/minecraft/class_8056 net/minecraft/item/trim/ArmorTrimPattern
+	FIELD field_42014 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_42015 REGISTRY_CODEC Lcom/mojang/serialization/Codec;
+	METHOD method_48446 getDescripting (Lnet/minecraft/class_6880;)Lnet/minecraft/class_2561;
```
Comment (haykam821) [inline mappings/net/minecraft/item/trim/ArmorTrimPattern.mapping]: Is this correct?

```suggestion
	METHOD method_48446 getDescription (Lnet/minecraft/class_6880;)Lnet/minecraft/class_2561;
```

File: mappings/net/minecraft/item/trim/ArmorTrimPatterns.mapping [hunk 4]
```diff
@@ -0,0 +1,16 @@
+CLASS net/minecraft/class_8057 net/minecraft/item/trim/ArmorTrimPatterns
+	METHOD method_48447 (Lnet/minecraft/class_1799;Lnet/minecraft/class_6880$class_6883;)Z
+		ARG 1 pattern
+	METHOD method_48448 get (Lnet/minecraft/class_5455;Lnet/minecraft/class_1799;)Ljava/util/Optional;
+		ARG 0 drm
```
Comment (haykam821) [inline mappings/net/minecraft/item/trim/ArmorTrimPatterns.mapping]:
```suggestion
		ARG 0 registryManager
```

File: mappings/net/minecraft/recipe/Recipe.mapping [hunk 5]
```diff
@@ -29,6 +29,7 @@ CLASS net/minecraft/class_1860 net/minecraft/recipe/Recipe
 		COMMENT
 		COMMENT <p>The returned stack should not be modified. To obtain the actual output,
 		COMMENT call {@link #craft(Inventory)}.
+		ARG 1 drm
```
Comment (haykam821) [inline mappings/net/minecraft/recipe/Recipe.mapping]:
```suggestion
		ARG 1 registryManager
```

File: mappings/net/minecraft/recipe/Recipe.mapping [hunk 6]
```diff
@@ -78,6 +79,7 @@ CLASS net/minecraft/class_1860 net/minecraft/recipe/Recipe
 		COMMENT @return the resulting item stack
 		ARG 1 inventory
 			COMMENT the input inventory
+		ARG 2 drm
```
Comment (haykam821) [inline mappings/net/minecraft/recipe/Recipe.mapping]:
```suggestion
		ARG 2 registryManager
```

File: mappings/net/minecraft/screen/ForagingScreenSlots.mapping [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	METHOD method_48364 builder ()Lnet/minecraft/class_8047$class_8048;
+	METHOD method_48365 has (I)Z
+		ARG 1 index
+	METHOD method_48366 getResult ()Lnet/minecraft/class_8047$class_8049;
+	METHOD method_48367 get (I)Lnet/minecraft/class_8047$class_8049;
+		ARG 1 index
+	METHOD method_48368 getInputs ()Ljava/util/List;
+	METHOD method_48369 getInputSlotCount ()I
+	METHOD method_48370 getInputSlotCount2 ()I
+	METHOD method_48371 getIndecies ()Ljava/util/List;
```
Comment (haykam821) [inline mappings/net/minecraft/screen/ForagingScreenSlots.mapping]:
```suggestion
	METHOD method_48371 getIndices ()Ljava/util/List;
```

File: mappings/net/minecraft/screen/ForagingScreenSlots.mapping [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+		ARG 2 resultSlot
+	METHOD method_48364 builder ()Lnet/minecraft/class_8047$class_8048;
+	METHOD method_48365 has (I)Z
+		ARG 1 index
+	METHOD method_48366 getResult ()Lnet/minecraft/class_8047$class_8049;
+	METHOD method_48367 get (I)Lnet/minecraft/class_8047$class_8049;
+		ARG 1 index
+	METHOD method_48368 getInputs ()Ljava/util/List;
+	METHOD method_48369 getInputSlotCount ()I
+	METHOD method_48370 getInputSlotCount2 ()I
```
Comment (haykam821) [inline mappings/net/minecraft/screen/ForagingScreenSlots.mapping]: What's the difference between these two methods? Usually, methods don't have numeric suffixes.

File: mappings/net/minecraft/screen/ForgingScreenHandler.mapping [hunk 9]
```diff
@@ -3,6 +3,8 @@ CLASS net/minecraft/class_4861 net/minecraft/screen/ForgingScreenHandler
 	FIELD field_22480 input Lnet/minecraft/class_1263;
 	FIELD field_22481 context Lnet/minecraft/class_3914;
 	FIELD field_22482 player Lnet/minecraft/class_1657;
+	FIELD field_41903 inputSlotIndecies Ljava/util/List;
```
Comment (haykam821) [inline mappings/net/minecraft/screen/ForgingScreenHandler.mapping]:
```suggestion
	FIELD field_41903 inputSlotIndices Ljava/util/List;
```

File: mappings/net/minecraft/item/trim/ArmorTrim.mapping [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+		ARG 1 drm
+		ARG 2 list
+	METHOD method_48427 equals (Lnet/minecraft/class_6880;Lnet/minecraft/class_6880;)Z
+		ARG 1 pattern
+		ARG 2 material
+	METHOD method_48428 getTrim (Lnet/minecraft/class_5455;Lnet/minecraft/class_1799;)Ljava/util/Optional;
+		ARG 0 drm
+		ARG 1 stack
+	METHOD method_48429 apply (Lnet/minecraft/class_5455;Lnet/minecraft/class_1799;Lnet/minecraft/class_8053;)Z
+		ARG 0 drm
```
Comment (haykam821) [inline mappings/net/minecraft/item/trim/ArmorTrim.mapping]:
```suggestion
		ARG 0 registryManager
```

File: mappings/net/minecraft/item/trim/ArmorTrim.mapping [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+	FIELD field_42001 genericModelIdGetter Ljava/util/function/Supplier;
+	METHOD <init> (Lnet/minecraft/class_6880;Lnet/minecraft/class_6880;)V
+		ARG 1 material
+		ARG 2 pattern
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 other
+	METHOD method_48424 getPattern ()Lnet/minecraft/class_6880;
+	METHOD method_48425 appendTooltip (Lnet/minecraft/class_1799;Lnet/minecraft/class_5455;Ljava/util/List;)V
+		ARG 0 stack
+		ARG 1 drm
```
Comment (haykam821) [inline mappings/net/minecraft/item/trim/ArmorTrim.mapping]:
```suggestion
		ARG 1 registryManager
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/data/server/recipe/TransformSmithingRecipeJsonBuilder.mapping [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+	METHOD method_48536 criterion (Ljava/lang/String;Lnet/minecraft/class_184;)Lnet/minecraft/class_8074;
+		ARG 1 name
+		ARG 2 conditions
+	METHOD method_48537 offerTo (Ljava/util/function/Consumer;Lnet/minecraft/class_2960;)V
+		ARG 1 exporter
+		ARG 2 id
+	METHOD method_48538 offerTo (Ljava/util/function/Consumer;Ljava/lang/String;)V
+		ARG 1 exporter
+		ARG 2 id
+	CLASS class_8075 TransfromSmithingRecipeJsonProvider
```
Comment (Shnupbups) [inline mappings/net/minecraft/data/server/recipe/TransformSmithingRecipeJsonBuilder.mapping]: Misspelling, plus the id is `smithing_transform`, would be best to keep that naming order

File: mappings/net/minecraft/screen/ForagingScreenSlots.mapping [hunk 13]
```diff
@@ -0,0 +1,35 @@
+CLASS net/minecraft/class_8047 net/minecraft/screen/ForagingScreenSlots
```
Comment (Shnupbups) [inline mappings/net/minecraft/screen/ForagingScreenSlots.mapping]: Misspelling

File: mappings/net/minecraft/data/server/recipe/TrimSmithingRecipeJsonBuilder.mapping [hunk 14]
```diff
@@ -0,0 +1,30 @@
+CLASS net/minecraft/class_8076 net/minecraft/data/server/recipe/TrimSmithingRecipeJsonBuilder
```
Comment (Shnupbups) [inline mappings/net/minecraft/data/server/recipe/TrimSmithingRecipeJsonBuilder.mapping]: The id is `smithing_trim`, would be best to keep that naming order

File: mappings/net/minecraft/recipe/TrimSmithingRecipe.mapping [hunk 15]
```diff
@@ -0,0 +1,14 @@
+CLASS net/minecraft/class_8062 net/minecraft/recipe/TrimSmithingRecipe
```
Comment (Shnupbups) [inline mappings/net/minecraft/recipe/TrimSmithingRecipe.mapping]: The id is `smithing_trim`, would be best to keep that naming order

File: mappings/net/minecraft/recipe/TransformSmithingRecipe.mapping [hunk 16]
```diff
@@ -0,0 +1,13 @@
+CLASS net/minecraft/class_8060 net/minecraft/recipe/TransformSmithingRecipe
```
Comment (Shnupbups) [inline mappings/net/minecraft/recipe/TransformSmithingRecipe.mapping]: The id is `smithing_transform`, would be best to keep that naming order

File: mappings/net/minecraft/data/server/recipe/TransformSmithingRecipeJsonBuilder.mapping [hunk 17]
```diff
@@ -0,0 +1,33 @@
+CLASS net/minecraft/class_8074 net/minecraft/data/server/recipe/TransformSmithingRecipeJsonBuilder
```
Comment (Shnupbups) [inline mappings/net/minecraft/data/server/recipe/TransformSmithingRecipeJsonBuilder.mapping]: The id is `smithing_transform`, would be best to keep that naming order

File: mappings/net/minecraft/data/server/recipe/TrimSmithingRecipeJsonBuilder.mapping [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+	METHOD method_48541 criterion (Ljava/lang/String;Lnet/minecraft/class_184;)Lnet/minecraft/class_8076;
+		ARG 1 name
+		ARG 2 conditions
+	METHOD method_48542 offerTo (Ljava/util/function/Consumer;Lnet/minecraft/class_2960;)V
+		ARG 1 exporter
+		ARG 2 id
+	METHOD method_48543 offerTo (Ljava/util/function/Consumer;Ljava/lang/String;)V
+		ARG 1 exporter
+		ARG 2 id
+	CLASS class_8077 TrimSmithingRecipeJsonProvider
```
Comment (Shnupbups) [inline mappings/net/minecraft/data/server/recipe/TrimSmithingRecipeJsonBuilder.mapping]: The id is `smithing_trim`, would be best to keep that naming order

File: mappings/net/minecraft/server/command/SpawnArmorTrimsCommand.mapping [hunk 19]
```diff
@@ -0,0 +1,17 @@
+CLASS net/minecraft/class_8044 net/minecraft/server/command/SpawnArmorTrimsCommand
+	FIELD field_41885 ITEMS Ljava/util/Map;
+	FIELD field_41886 PATTERNS Ljava/util/List;
+	FIELD field_41887 MATERIALS Ljava/util/List;
+	FIELD field_41888 PATTERN_GETTER Ljava/util/function/ToIntFunction;
+	FIELD field_41889 TRIM_GETTER Ljava/util/function/ToIntFunction;
```
Comment (Shnupbups) [inline mappings/net/minecraft/server/command/SpawnArmorTrimsCommand.mapping]: These get their indexes in the lists, not the patterns and trims themselves.

File: mappings/net/minecraft/data/server/recipe/TrimSmithingRecipeJsonBuilder.mapping [hunk 20]
```diff
@@ -0,0 +1,30 @@
+CLASS net/minecraft/class_8076 net/minecraft/data/server/recipe/TrimSmithingRecipeJsonBuilder
+	FIELD field_42097 category Lnet/minecraft/class_7800;
+	FIELD field_42098 template Lnet/minecraft/class_1856;
+	FIELD field_42099 base Lnet/minecraft/class_1856;
+	FIELD field_42100 addition Lnet/minecraft/class_1856;
+	FIELD field_42101 advancement Lnet/minecraft/class_161$class_162;
+	FIELD field_42102 serilizer Lnet/minecraft/class_1865;
```
Comment (Shnupbups) [inline mappings/net/minecraft/data/server/recipe/TrimSmithingRecipeJsonBuilder.mapping]: Misspelling

File: mappings/net/minecraft/client/gui/screen/ingame/ForgingScreen.mapping [hunk 21]
```diff
@@ -11,3 +11,7 @@ CLASS net/minecraft/class_4894 net/minecraft/client/gui/screen/ingame/ForgingScr
 		ARG 3 mouseY
 		ARG 4 delta
 	METHOD method_25445 setup ()V
+	METHOD method_48467 drawBackgroundExtras (Lnet/minecraft/class_4587;II)V
```
Comment (Shnupbups) [inline mappings/net/minecraft/client/gui/screen/ingame/ForgingScreen.mapping]: This is specifically for drawing the arrow with a cross through it for when the recipe is invalid

File: mappings/net/minecraft/item/ArmorItem.mapping [hunk 22]
```diff
@@ -1,18 +1,31 @@
...
+	METHOD method_48397 (Ljava/util/EnumMap;)V
+		ARG 0 map
+	METHOD method_48398 getSlot ()Lnet/minecraft/class_1738$class_8051;
 	METHOD method_7684 dispenseArmor (Lnet/minecraft/class_2342;Lnet/minecraft/class_1799;)Z
 		ARG 0 pointer
 		ARG 1 armor
 	METHOD method_7685 getSlotType ()Lnet/minecraft/class_1304;
 	METHOD method_7686 getMaterial ()Lnet/minecraft/class_1741;
 	METHOD method_7687 getProtection ()I
+	CLASS class_8051 ArmorSlot
```
Comment (Shnupbups) [inline mappings/net/minecraft/item/ArmorItem.mapping]: This is not a slot, it's a type of armor

File: mappings/net/minecraft/item/trim/ArmorTrimMaterials.mapping [hunk 23]
```diff
@@ -0,0 +1,9 @@
+CLASS net/minecraft/class_8055 net/minecraft/item/trim/ArmorTrimMaterials
+	METHOD method_48441 makeKey (Ljava/lang/String;)Lnet/minecraft/class_5321;
```
Comment (Shnupbups) [inline mappings/net/minecraft/item/trim/ArmorTrimMaterials.mapping]: Usually we just call these `of`. Also why is the parameter not mapped?

File: mappings/net/minecraft/item/trim/ArmorTrimPatterns.mapping [hunk 24]
```diff
@@ -0,0 +1,16 @@
+CLASS net/minecraft/class_8057 net/minecraft/item/trim/ArmorTrimPatterns
+	METHOD method_48447 (Lnet/minecraft/class_1799;Lnet/minecraft/class_6880$class_6883;)Z
+		ARG 1 pattern
+	METHOD method_48448 get (Lnet/minecraft/class_5455;Lnet/minecraft/class_1799;)Ljava/util/Optional;
+		ARG 0 registryManager
+		ARG 1 stack
+	METHOD method_48449 makeKey (Ljava/lang/String;)Lnet/minecraft/class_5321;
```
Comment (Shnupbups) [inline mappings/net/minecraft/item/trim/ArmorTrimPatterns.mapping]: Usually we just call these `of`

File: mappings/net/minecraft/item/trim/ArmorTrimPatterns.mapping [hunk 25]
```diff
@@ -0,0 +1,16 @@
...
+		ARG 0 name
```
Comment (Shnupbups) [inline mappings/net/minecraft/item/trim/ArmorTrimPatterns.mapping]: This should be `id`

**Review by Shnupbups — DISMISSED** [review]

File: mappings/net/minecraft/item/ArmorMaterials.mapping [hunk 26]
```diff
@@ -1,5 +1,9 @@
 CLASS net/minecraft/class_1740 net/minecraft/item/ArmorMaterials
 	FIELD field_21978 knockbackResistance F
+	FIELD field_41941 CODEC Lnet/minecraft/class_3542$class_7292;
+	FIELD field_41942 BASE_DURABILITY Ljava/util/EnumMap;
+	FIELD field_41943 protectionAmounts Ljava/util/EnumMap;
+	FIELD field_41944 isTrimmable Z
```
Comment (apple502j) [inline mappings/net/minecraft/item/ArmorMaterials.mapping]: `trimmable`?

File: mappings/net/minecraft/item/ArmorMaterials.mapping [hunk 27]
```diff
@@ -10,8 +14,26 @@ CLASS net/minecraft/class_1740 net/minecraft/item/ArmorMaterials
 	METHOD <init> (Ljava/lang/String;ILjava/lang/String;ILjava/util/EnumMap;ILnet/minecraft/class_3414;FFLjava/util/function/Supplier;Z)V
 		ARG 3 name
 		ARG 4 durabilityMultiplier
+		ARG 5 protectionAmounts
 		ARG 6 enchantability
 		ARG 7 equipSound
 		ARG 8 toughness
 		ARG 9 knockbackResistance
 		ARG 10 repairIngredientSupplier
+		ARG 11 isTrimmable
```
Comment (apple502j) [inline mappings/net/minecraft/item/ArmorMaterials.mapping]: same

File: mappings/net/minecraft/item/ArmorItem.mapping [hunk 28]
```diff
@@ -1,18 +1,31 @@
 CLASS net/minecraft/class_1738 net/minecraft/item/ArmorItem
 	FIELD field_21976 knockbackResistance F
 	FIELD field_23741 attributeModifiers Lcom/google/common/collect/Multimap;
+	FIELD field_41933 slot Lnet/minecraft/class_1738$class_8051;
```
Comment (apple502j) [inline mappings/net/minecraft/item/ArmorItem.mapping:4]: Wrong name, but shnupbups PR fixes it

File: mappings/net/minecraft/item/trim/ArmorTrim.mapping [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+	FIELD field_41997 TRIM_SECTION_TEXT Lnet/minecraft/class_2561;
+	FIELD field_41998 material Lnet/minecraft/class_6880;
+	FIELD field_41999 pattern Lnet/minecraft/class_6880;
+	FIELD field_42000 leggingsModelIdGetter Ljava/util/function/Supplier;
+	FIELD field_42001 genericModelIdGetter Ljava/util/function/Supplier;
+	METHOD <init> (Lnet/minecraft/class_6880;Lnet/minecraft/class_6880;)V
+		ARG 1 material
+		ARG 2 pattern
+	METHOD equals (Ljava/lang/Object;)Z
+		ARG 1 other
```
Comment (apple502j) [inline mappings/net/minecraft/item/trim/ArmorTrim.mapping:14]: `o`, but we can fix later

**Review by apple502j — APPROVED** [review]
Issues to be fixed in another PR, let's merge the base

**Pull request merged**